extensions/wikia/MonetizationModule/tests/MonetizationModuleTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+class MonetizationModuleTest extends WikiaBaseTest {
+
+	const TEST_CITY_ID = 79860;
+
+	protected $orgSkin;
+
+	public function setUp() {
+		$this->setupFile = dirname( __FILE__ ).'/../MonetizationModule.setup.php';
+		parent::setUp();
+
+		$this->mockCache();
+	}
+
+	protected function mockCache( $value = null ) {
+		$mock_cache = $this->getMock( 'stdClass', [ 'get', 'set', 'delete' ] );
+		$mock_cache->expects( $this->any() )
+			->method('get')
+			->will( $this->returnValue( $value ) );
+		$mock_cache->expects( $this->any() )
+			->method( 'set' );
+		$mock_cache->expects( $this->any())
+			->method( 'delete' );
+
+		$this->mockGlobalVariable( 'wgMemc', $mock_cache );
+		$this->mockGlobalVariable( 'wgCityId', self::TEST_CITY_ID );
+	}
+
+	protected function setUpSkin( $skinName ) {
+		$skin = Skin::newFromKey( $skinName );
+		$this->orgSkin = RequestContext::getMain()->getSkin();
+		RequestContext::getMain()->setSkin( $skin );
+	}
+
+	protected function tearDownSkin() {
+		RequestContext::getMain()->setSkin( $this->orgSkin );
+	}
+
+	protected function mockUser( $isAnon = true ) {
+		$user = $this->getMock( 'User', [ 'isAnon' ] );
+		$user->expects( $this->any() )
+			->method( 'isAnon' )
+			->will( $this->returnValue( $isAnon ) );
+		$this->mockGlobalVariable( 'wgUser', $user );
+	}
+
+	protected function mockTitle( $pageExists = true, $isMainPage = false, $namespace = 0 ) {
+		$title = $this->getMock( 'Title', [ 'exists', 'isMainPage', 'getNamespace' ] );
+		$title->expects( $this->any() )
+			->method( 'exists' )
+			->will( $this->returnValue( $pageExists ) );
+		$title->expects( $this->any() )
+			->method( 'isMainPage' )
+			->will( $this->returnValue( $isMainPage ) );
+		$title->expects( $this->any() )
+			->method( 'getNamespace' )
+			->will( $this->returnValue( $namespace ) );
+
+		return $title;
+	}
+
+	/**
+	 * @dataProvider canShowModuleDataProvider
+	 */
+	public function testCanShowModule( $corporatePage, $pageExists, $mainPage, $ns, $actionPage, $skin, $anon, $exp ) {
+		$this->setUpSkin( $skin );
+
+		$this->mockGlobalVariable( 'wgContentNamespaces', [ 0 ] );
+
+		$this->mockStaticMethod( 'WikiaPageType', 'isCorporatePage', $corporatePage );
+		$this->mockStaticMethod( 'WikiaPageType', 'isActionPage', $actionPage );
+
+		$this->mockUser( $anon );
+		$title = $this->mockTitle(  $pageExists, $mainPage, $ns );
+
+		$result = MonetizationModuleHelper::canShowModule( $title );
+		$this->assertEquals( $exp, $result );
+
+		$this->tearDownSkin();
+	}
+
+	public function canShowModuleDataProvider() {
+		return [
+			// Module shown - article page
+			[ false, true, false, NS_MAIN, false, 'oasis', true, true ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "oasis" 10 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)